kernel/sched.c

```diff
@@ -1065,17 +1068,19 @@ void z_sched_abort(struct k_thread *thread)
 	 * running on or because we caught it idle in the queue
 	 */
 	while ((thread->base.thread_state & _THREAD_DEAD) == 0U) {
-		LOCKED(&sched_spinlock) {
-			if (z_is_thread_prevented_from_running(thread)) {
-				__ASSERT(!z_is_thread_queued(thread), "");
-				thread->base.thread_state |= _THREAD_DEAD;
-			} else if (z_is_thread_queued(thread)) {
-				_priq_run_remove(&_kernel.ready_q.runq, thread);
-				z_mark_thread_as_not_queued(thread);
-				thread->base.thread_state |= _THREAD_DEAD;
-			} else {
-				k_busy_wait(100);
-			}
+		key = k_spin_lock(&sched_spinlock);
+		if (z_is_thread_prevented_from_running(thread)) {
+			__ASSERT(!z_is_thread_queued(thread), "");
+			thread->base.thread_state |= _THREAD_DEAD;
+			k_spin_unlock(&sched_spinlock, key);
+		} else if (z_is_thread_queued(thread)) {
+			_priq_run_remove(&_kernel.ready_q.runq, thread);
+			z_mark_thread_as_not_queued(thread);
+			thread->base.thread_state |= _THREAD_DEAD;
+			k_spin_unlock(&sched_spinlock, key);
+		} else {
+			k_spin_unlock(&sched_spinlock, key);
+			k_busy_wait(100);
```

Comment:
Style wise, I prefer to keep the lock and unlock pairs at the same level of indentation.  Can you change this to something like:

```c
    k_spin_unlock(&sched_spinlock, key);
    k_busy_wait(100);
    k_spin_lock(&sched_spinlock);
```

Reply:
I did't find a way to keep the lock and unlock parirs at the same level.  In the provided codes, k_spin_lock cannot be after k_busy_wait  as while((thread->base.thread & _THREAD_DEAD)) may exit before call k_spin_unlock.   How about the following:

```
		LOCKED(&sched_spinlock) {
			if (z_is_thread_prevented_from_running(thread)) {
				__ASSERT(!z_is_thread_queued(thread), "");
				thread->base.thread_state |= _THREAD_DEAD;
			} else if (z_is_thread_queued(thread)) {
				_priq_run_remove(&_kernel.ready_q.runq, thread);
				z_mark_thread_as_not_queued(thread);
				thread->base.thread_state |= _THREAD_DEAD;
			}
		}
                k_busy_wait(100);
```
But the problem is that k_busy_wait(100) will be always executed